src/generate/codemodel/CodeModelAzImpl.ts

```diff
@@ -869,7 +869,15 @@ export class CodeModelCliImpl implements CodeModelAz {
                     ) {
                         action['constants'][`'${keyToMatch}'`] = `'${valueToMatch}'`;
                     } else {
+                        let parentKeys = [];
+                        if (!isNullOrUndefined(tmpParam['flattenedNames'])) {
+                            parentKeys = tmpParam['flattenedNames'].map((fn) => ToSnakeCase(fn));
```

Comment:
not sure whether ToSnakeCase can fit all rare cases. 
Go through all ancestor levels' pythonName should be better.

Reply:
They are not the same. we can't use it like this.